util/internal/integrationtest/mapper/orm/src/main/java/org/hibernate/search/util/impl/integrationtest/mapper/orm/OrmSetupHelper.java

```diff
@@ -208,6 +217,34 @@ protected void close(SessionFactory toClose) {
 		}
 	}
 
+	@Override
+	public <T> T interceptTestClassConstructor(Invocation<T> invocation,
+			ReflectiveInvocationContext<Constructor<T>> invocationContext, ExtensionContext extensionContext)
+			throws Throwable {
+		T testInstance = InvocationInterceptor.super.interceptTestClassConstructor( invocation, invocationContext,
+				extensionContext
+		);
+
+		this.ormSetupHelperCleaner = new OrmSetupHelperCleaner( testInstance, invocationContext.getTargetClass() );
+		return testInstance;
+	}
+
+	@Override
+	public void interceptTestTemplateMethod(Invocation<Void> invocation,
+			ReflectiveInvocationContext<Method> invocationContext, ExtensionContext extensionContext)
+			throws Throwable {
+		invocation.proceed();
+		this.ormSetupHelperCleaner.cleanupData( sessionFactory );
+	}
+
+	@Override
+	public void interceptTestMethod(Invocation<Void> invocation, ReflectiveInvocationContext<Method> invocationContext,
+			ExtensionContext extensionContext)
+			throws Throwable {
+		invocation.proceed();
+		this.ormSetupHelperCleaner.cleanupData( sessionFactory );
+	}
```

Comment:
Isn't this equivalent to an `AfterEachCallback`?

Reply:
yes, that's mostly the same. I've been using this invocation interceptor as initially I was looking for the "configuration parameters" that were used in the junit4 rule to detect if the changes require data cleanup or recreating the session and only `invocationContext` had that info accessible without resorting to reflection. But with that gone (comparing parameters) I'll move these to afterEach.